client/src/pages/_app.tsx

```diff
@@ -27,7 +25,8 @@ type AppPropsWithLayout = AppProps & {
 };
 
 const MyApp: NextPage<AppPropsWithLayout> = ({ Component, pageProps }) => {
-  const store = useMemo(() => initStore(pageProps.query), [pageProps.query]);
+  const [store] = useState(() => initStore(pageProps.query));
+  const [queryClient] = useState(() => new QueryClient());
```

Comment:
this change should be included in the changelog, add a comment explaining why it was moving inside